mlflow/pyfunc/__init__.py

```diff
@@ -720,7 +721,22 @@ def _update_dependencies_schemas_in_prediction_context(self):
                 }
             )
 
+    @contextmanager
+    def _try_get_or_generate_prediction_context(self):
+        # set context for prediction if it's not set
+        # NB: in model serving the prediction context must be set
+        # with a request_id
+        if (context := _try_get_prediction_context()) is None:
+            context = Context()
+        with set_prediction_context(context):
+            yield context
+
     def predict(self, data: PyFuncInput, params: Optional[Dict[str, Any]] = None) -> PyFuncOutput:
+        with self._try_get_or_generate_prediction_context() as context:
+            self._update_dependencies_schemas_in_prediction_context(context)
+            return self._predict(data, params)
```

Comment:
Do we want to create a context even if users don't have tracing related code while using pyfunc? Would these cause any empty trace info with tags being created?

Reply:
If user doesn't trace their code, then no traces will be logged at all. I can add a test case for this :)